File: students/students/src/MON.SPA/src/models/studentClass/classGroup.js
export class ClassGroup{
    constructor(obj = {}){
        this.schoolYear = obj.schoolYear || 0;
        this.institutionId = obj.institutionId || 0;
        this.institutionName = obj.institutionName || '';
        this.className = obj.className || '';
        this.basicClassName = obj.basicClassName || '';
        this.classTypeName = obj.classTypeName || '';
        this.classSpecialityName = obj.classSpecialityName || '';
        this.classEduFormName = obj.classEduFormName || '';
    }
}

File: school-books/backend/SB.Domain/External/InstYear/Entities/ClassGroup.cs
namespace SB.Domain;

public class ClassGroup
{
    // EF constructor
    private ClassGroup()
    {
        this.ClassName = null!;
    }

    // only used properties should be mapped

    public int ClassId { get; private set; }

    public int SchoolYear { get; private set; }

    public int InstitutionId { get; private set; }

    public string ClassName { get; private set; }

    public string? ParalellClassName { get; private set; }

    public int? ParentClassId { get; private set; }

    public int? BasicClassId { get; private set; }

    public int? ClassTypeId { get; private set; }

    public bool? IsCombined { get; private set; }

    public bool? IsSpecNeed { get; private set; }

    public bool? IsNotPresentForm { get; private set; }

    public int? ClassSpecialityId { get; private set; }

    public bool IsValid { get; private set; }
}
